staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go

```diff
@@ -230,7 +230,41 @@ func (h *UpgradeAwareHandler) ServeHTTP(w http.ResponseWriter, req *http.Request
 	proxy := httputil.NewSingleHostReverseProxy(&url.URL{Scheme: h.Location.Scheme, Host: h.Location.Host})
 	proxy.Transport = h.Transport
 	proxy.FlushInterval = h.FlushInterval
-	proxy.ServeHTTP(w, newReq)
+	proxy.ServeHTTP(newFlushHeaderWriter(w), newReq)
+}
+
+// TODO(liggitt): drop this once https://github.com/golang/go/issues/31125 is fixed
+func newFlushHeaderWriter(w http.ResponseWriter) http.ResponseWriter {
+	flusher, ok := w.(http.Flusher)
+	if !ok {
+		return w
```

Comment:
if http.Flusher is not implemented by the underlying responseWriter, we silently return the writer unmodified. Thus, it lies about what it does (newFlushHeaderWriter can return a non-flushing response writer).

Either we are ok with that (and should rename the fuction accorodingly), or we should return an error in this cas.